### PR TITLE
Refactor code to new support wait and request calls

### PR DESCRIPTION
**Pull request by victorcolombo**
Resolves https://github.com/kurtosis-tech/eth2-package/issues/34
Resolves https://github.com/kurtosis-tech/kurtosis/issues/619

### CODE REVIEW

File: src/participant_network/cl/cl_node_health_checker.star [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# this is a dictionary as this will get serialzed to JSON
+def wait(service_id, port_id):
+	recipe = struct(
+        service_id = service_id,
+        method= "GET",
+        endpoint = "/eth/v1/node/health",
+        content_type = "application/json",
+        port_id = port_id
+    )
+	return wait(recipe, "code", "IN", [200, 206, 503])
```
Comment (h4ck3rk3y) [inline src/participant_network/cl/cl_node_health_checker.star:10]: lets add a new line

File: src/participant_network/cl/cl_node_health_checker.star [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# this is a dictionary as this will get serialzed to JSON
```
Comment (h4ck3rk3y) [inline src/participant_network/cl/cl_node_health_checker.star:1]: old comment

File: main.star [hunk 3]
```diff
@@ -77,9 +77,17 @@ def run(args):
...
+				service_id = first_cl_client_id,
+				method= "GET",
+				endpoint = "/eth/v1/beacon/states/head/finality_checkpoints",
+				content_type = "application/json",
+				port_id = HTTP_PORT_ID_FOR_FACT,
+				extract = {
+					"finalized_epoch": ".data.finalized.epoch"
+				}
+			)
+			wait(epoch_recipe, "finalized_epoch", ">", 0)
```
Comment (h4ck3rk3y) [inline main.star:90]: did you run this with waitForEpochFinalization turned to true?

File: src/participant_network/cl/lighthouse/lighthouse_launcher.star [hunk 4]
```diff
@@ -99,8 +99,14 @@ def launch(
 	beacon_service = add_service(beacon_node_service_id, beacon_config)
 
 	# TODO check whether its 200, 206 or 503 like golang
-	define_fact(service_id = beacon_node_service_id, fact_name = BEACON_HEALTH_FACT_NAME, fact_recipe = struct(method= "GET", endpoint = "/eth/v1/node/health", content_type = "application/json", port_id = BEACON_HTTP_PORT_ID))
-	wait(service_id = beacon_node_service_id, fact_name = BEACON_HEALTH_FACT_NAME)
+	health_recipe = struct(
```
Comment (h4ck3rk3y) [inline src/participant_network/cl/lighthouse/lighthouse_launcher.star:102]: could we use src/participant_network/cl/cl_node_health_checker.star here?

File: src/participant_network/cl/lighthouse/lighthouse_launcher.star [hunk 5]
```diff
@@ -120,8 +126,17 @@ def launch(
...
+		service_id = beacon_node_service_id,
+		method= "GET",
+		endpoint = "/eth/v1/node/identity",
+		content_type = "application/json",
+		port_id = BEACON_HTTP_PORT_ID,
+		extract = {
+			"enr": ".data.enr"
+		}
+	)
+	beacon_node_enr = request(beacon_node_identity_recipe)["extract.enr"]
```
Comment (h4ck3rk3y) [inline src/participant_network/cl/lighthouse/lighthouse_launcher.star:139]: neat!

File: src/participant_network/el/besu/besu_launcher.star [hunk 6]
```diff
@@ -62,13 +60,23 @@ def launch(
...
+		method= "POST",
+		endpoint = "",
+		body = '{"method":"admin_nodeInfo","params":[],"id":1,"jsonrpc":"2.0"}',
+		content_type = "application/json",
+		port_id = RPC_PORT_ID,
+		extract = {
+			"enode": ".result.enode",
+		}
+	)
+	response = wait(result_recipe, "extract.enode", "!=", "")
```
Comment (h4ck3rk3y) [inline src/participant_network/el/besu/besu_launcher.star:74]: neat

**Review by h4ck3rk3y — APPROVED** [review]
looks good, have you tried the wait for epoch ?

File: src/participant_network/cl/lighthouse/lighthouse_launcher.star (hunk 4 above)
Comment (victorcolombo) [reply src/participant_network/cl/lighthouse/lighthouse_launcher.star:102]: good catch

File: main.star [hunk 7]
```diff
@@ -77,9 +77,17 @@ def run(args):
 			print("Waiting for the first finalized epoch")
 			first_cl_client = all_cl_client_contexts[0]
 			first_cl_client_id = first_cl_client.beacon_service_id
-			define_fact(service_id = first_cl_client_id, fact_name = FIRST_NODE_FINALIZATION_FACT, fact_recipe = struct(method= "GET", endpoint = "/eth/v1/beacon/states/head/finality_checkpoints", content_type = "application/json", port_id = HTTP_PORT_ID_FOR_FACT, field_extractor = ".data.finalized.epoch"))
-			finalized_epoch = wait(service_id = first_cl_client_id, fact_name = FIRST_NODE_FINALIZATION_FACT)
-			# TODO make an assertion on the finalized_epoch > 0
+			epoch_recipe = struct(
```
Comment (h4ck3rk3y) [inline main.star:80]: can you pass a time out as this sometimes needs 32 minutes

Comment (h4ck3rk3y) [reply main.star:80]: well actually there's some math we can do to calculate that https://github.com/kurtosis-tech/eth2-merge-kurtosis-module/blob/master/kurtosis-module/impl/module.go#L229-L262

**Review by h4ck3rk3y — CHANGES_REQUESTED** [review]

File: main.star (hunk 3 above)
Comment (h4ck3rk3y) [reply main.star:90]:
```
{
  "participants": [
    {
      "el_client_type": "geth",
      "el_client_image": "ethereum/client-go:v1.10.26",
      "cl_client_type": "lighthouse",
      "cl_client_image": "sigp/lighthouse:v3.3.0"
    },
    {
      "el_client_type": "geth",
      "el_client_image": "ethereum/client-go:v1.10.26",
      "cl_client_type": "prysm",
      "cl_client_image": "gcr.io/prysmaticlabs/prysm/beacon-chain:capella-post-merge,gcr.io/prysmaticlabs/prysm/validator:capella-post-merge"
    }
  ],
  "network_params": {
    "network_id": "3151908",
    "deposit_contract_address": "0x4242424242424242424242424242424242424242",
    "seconds_per_slot": 12,
    "slots_per_epoch": 32,
    "num_validator_keys_per_node": 64,
    "preregistered_validator_keys_mnemonic": "giant issue aisle success illegal bike spike question tent bar rely arctic volcano long crawl hungry vocal artwork sniff fantasy very lucky have athlete"
  },
  "wait_for_finalization": true,
  "wait_for_verifications": false,
  "global_client_log_level": "info"
}
```

Comment (victorcolombo) [reply main.star:90]: Doing it rn

Comment (victorcolombo) [reply main.star]: Worked!

File: main.star (hunk 7 above)
Comment (h4ck3rk3y) [reply main.star:80]: that sounds good

File: src/participant_network/cl/cl_node_health_checker.star (hunk 1 above)
Comment (h4ck3rk3y) [reply src/participant_network/cl/cl_node_health_checker.star:9]: the new line isn't there

**Pull request merged**